src/pages/visualizer/VisualizerProvider.tsx

```diff
@@ -119,67 +79,75 @@ class Visualizer extends Component<VisualizerProps, VisualizerState> {
       } else {
         browser = 'your browser'
       }
-      this.setState({ toastText: this.messages(browser).text })
+      setToastText(messages(browser).notSupported)
     }
+  }, [])
 
-    this.updateVisibility()
-    this.updateAudioStream()
-    this.updateDominantColors()
+  if(!webGLExists) {
+    return null
   }
 
-  componentWillUnmount() {
-    Visualizer1?.stop()
-  }
+  // Update Colors
+  useEffect(() => {
+    if(dominantColors !== null) {
+      Visualizer1?.setDominantColors(dominantColors)
+    }
+  }, [isVisible, currentQueueItem])
+
+  // Rebind audio
+  useEffect(() => {
```

Comment:
one thought is that we could create custom hooks with the name like `useRebindAudio`

Reply:
Yeah I think that'd be a good idea if this logic was re-used but this is the only place this particular effect exists.

Reply:
I think I generally agree w/ not breaking out a hook into its own useX unless it's 
1) super complicated
2) shared w/ other components

I do think comments for the useEffects are useful though

Reply:
That all makes sense. Comments are great, agreed